Jellyfin.Server/Startup.cs

```diff
@@ -0,0 +1,81 @@
+using Jellyfin.Server.Extensions;
+using MediaBrowser.Controller;
+using MediaBrowser.Controller.Configuration;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Jellyfin.Server
+{
+    /// <summary>
+    /// Startup configuration for the Kestrel webhost.
+    /// </summary>
+    public class Startup
+    {
+        private readonly IServerConfigurationManager _serverConfigurationManager;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Startup" /> class.
+        /// </summary>
+        /// <param name="serverConfigurationManager">The server configuration manager.</param>
+        public Startup(IServerConfigurationManager serverConfigurationManager)
+        {
+            _serverConfigurationManager = serverConfigurationManager;
+        }
+
+        /// <summary>
+        /// Configures the service collection for the webhost.
+        /// </summary>
+        /// <param name="services">The service collection.</param>
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddResponseCompression();
+            services.AddHttpContextAccessor();
+            services.AddJellyfinApi(_serverConfigurationManager.Configuration.BaseUrl.TrimStart('/'));
+
+            services.AddJellyfinApiSwagger();
+
+            // configure custom legacy authentication
+            services.AddCustomAuthentication();
+
+            services.AddJellyfinApiAuthorization();
```

Comment:
Shouldn't these things go first? So if something tries to access JF while it's starting up it would be protected faster

Reply:
This is only service registration.